src/util/config.cpp

```diff
@@ -653,12 +653,16 @@ configt::ansi_ct::c_standardt configt::ansi_ct::default_c_standard()
   // By default, clang on FreeBSD builds C code in GNU C99
   return c_standardt::C99;
   #else
-  return c_standardt::C99;
+  // By default, gcc 5.4 or higher use gnu11; older versions use gnu89
```

Comment:
I continue to think we should parse the version of the compiler we're using, and vary our behaviour for compatibility (previously suggested to avoid defining _Float32 et al except when we're pretending to be GCC 7+)

Reply:
@smowton my experience with armcc (and that of EDG, the front-end it used) was that when we add GCC emulation it proved to be quite important for customers that we respected the GCC version we were pretending to be - a surprising amount of software broke if we didn't (and in some cases, this included needing to emulate certain GCC bugs).

Reply:
The version parsing is now in #2240.